extensions/notebook/src/book/bookTreeView.ts

```diff
@@ -785,5 +801,52 @@ export class BookTreeViewProvider implements vscode.TreeDataProvider<BookTreeIte
 		return localRoots;
 	}
 
+	public async undo(): Promise<void> {
+		if (this.undoTocFiles?.length > 0 && this.undoMovedFiles?.length > 0) {
+			let files = this.undoMovedFiles.pop();
+			this.redoMovedFiles.push(files);
+			this.redoTocFiles.push(this.undoTocFiles.pop());
```

Comment:
You're only removing one TOC entry - but in the code above you're spreading the tocFiles array before pushing it - so if a book has multiple TOC entries then you'll only be removing one. Is that intended? And if so why don't you treat movedFiles the same?